src/test/java/com/es/phoneshop/web/ProductPriceHistoryPageServletTest.java

```diff
@@ -0,0 +1,87 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.dao.ProductDao;
+import com.es.phoneshop.exception.ProductNotFoundException;
+import com.es.phoneshop.model.product.Product;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProductPriceHistoryPageServletTest {
```

Comment:
You probably might let this test class only verify that jsp path passed to getRequestDispatcher is correct for price history page. The rest of the logic is already covered by ProductDetailsPageServletTest